### DISCUSSION TITLE
Creating a context menu - Vanilla JS

### DISCUSSION THREAD

**Discussion by vdawg-git** [floating-ui/floating-ui#1916] [Q&A] [answered]
How can I implement a context menu with Floating UI?
I am using Svelte, but if I know how to do this with vanilla JS, I should be able to convert it.

The documentation gives a React example, but I do not understand it (I dont know React :D )

Furthermore, I want to use the library to create normal dropdown menus.

Edit: So, I figured out that I will need to create my own platform for that. However, half of the code I would need to write is within `@floating-ui/dom`, but the functions are not exported. Maybe exporting them would make it easier to write a platform for custom use cases?

**Comment by vdawg-git** [comment, marked as answer]
For anyone who wants to do this, I found a solution.
Just pass in a fake HTMLElement to `computePosition`, like this:
 ```javascript
 const { clientX, clientY } = cursorLocation

computePosition(
  {
    clientHeight: 1,
    clientWidth: 1,
    clientLeft: x,
    clientTop: y,
    getBoundingClientRect: () => ({
      x: clientX,
      y: clientY,
      top: clientY,
      bottom: 0,
      left: clientX,
      right: 0,
      width: 1,
      height: 1,
    }),
  },
  floatingElement,
)
// .then( doSomething )
```

**Comment by vaynevayne** [comment]
hi, where is react demo ?
